### PR TITLE
Extraction of software components from dockerfiles

### PR DESCRIPTION
**Pull request by Prebzs**
Added a project that can detect and extract software components from dockerfiles. Dockerfiles can be crawled, a crawler for GitHub is implemented. Extracted components can be retrieved as tosca model

- [x] Ensure that you followed http://eclipse.github.io/winery/dev/ToolChain#github---prepare-pull-request. Especially, we require **a single commit**
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files
- [x] Ensure that you appear in `NOTICE` at Copyright Holders
- [x] Tests created for changes
- [x] Documentation updated (if needed)
- [x] Screenshots added (for UI changes)

### CODE REVIEW

**Comment by genie-winery** [comment]
Can one of the admins verify this patch?

File: org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/fileanalyzer/CommandAnalyzer/ApkAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.List;
+
+public class ApkAnalyzer implements CommandAnalyzer {
+    public List<Component> analyze(String command) {
+        if (!command.contains(" add ")) {
+            return new ArrayList<>();
+        }
+        command = command.replace(Commands.Apk.asString(), "");
+        String[] words = command.split("\\s");
+        List<String> packets = new ArrayList<>();
```
Comment (Vogel612) [inline org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/fileanalyzer/CommandAnalyzer/ApkAnalyzer.java]: the correct term here would be "tokens"

File: org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/crawler/Crawler.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            while (crawledDockerfilesCount < number) {
+                List<Dockerfile> newDockerfiles = crawlerInstance.crawlDockerfiles();
+                crawledDockerfilesCount += newDockerfiles.size();
+                synchronized (crawledDockerfiles) {
+                    crawledDockerfiles.addAll(newDockerfiles);
+                }
+                for (Dockerfile dockerfile : newDockerfiles) {
+                    String filePath = dockerfile.getRepoName() + "." + dockerfile.getPath();
+                    filePath = filePath.replace("/", ".");
+                    filePath = System.getProperty("user.home") + "/Dockerfiles/" + filePath;
```
[the remark this answers is not in the document]
Comment (Prebzs) [reply org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/crawler/Crawler.java]: I'm not aware of any working directory that could be used here, but using home directory is also not the best solution. I change it to a parameter.

File: org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/fileanalyzer/CommandAnalyzer/ApkAnalyzer.java (hunk 1 above)
Comment (Prebzs) [reply org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/fileanalyzer/CommandAnalyzer/ApkAnalyzer.java]: I disagree. But packets is also wrong. In this context the extracted parts of the analyzed commands are software packages. I rename it to packages in all analyzer-classes.

File: org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/DeployableComponents.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        try {
+            Thread.sleep(3600000);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        comp.stop();
+
+        HashMap<Component, List<Pair<Component, Integer>>> results = comp.getFoundComponents();
+ */
```
[the remark this answers is not in the document]
Comment (Prebzs) [reply org.eclipse.winery.tools.deployablecomponents/src/main/java/org/eclipse/winery/tools/deployablecomponents/DeployableComponents.java]: Forgot to remove it here. It is already added to docs/user/

**Comment by lharzenetter** [comment]
Filed commit 1cc1eb6 as [CQ 20473](https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20473)

**Pull request merged**